packages/components/pie-assistive-text/src/index.ts

```diff
@@ -46,12 +35,12 @@ export class PieAssistiveText extends LitElement implements AssistiveTextProps {
             class="c-assistiveText"
             data-test-id="pie-assistive-text"
             variant=${ifDefined(variant)}>
-            ${this.renderIcon()}
-            <slot></slot>
-        </p>`;
+                ${variant === 'success' ? html`<icon-check-circle class="c-assistiveText-icon" size="s"></icon-check-circle>` : nothing}
```

Comment:
I understand `variant` can only be one thing, either success or error, however this doesn't seem the best pattern to solve the problem as it can become prone to inconsistency.

Can we solve this with something like this?

```
const iconTag = variant === 'success' ? 'icon-check-circle' : 'icon-alert-circle';

return html`<${iconTag} class="c-assistiveText-icon" size="s"></${iconTag}>`
```

Reply:
I did try this, but it's not actually string interpolation as far as I can tell.

I have however refactored to follow Lit recommendations for conditional rendering - https://lit.dev/docs/templates/conditionals/#conditionals-with-if-statements

Reply:
Just updated as part Lit recommendations for conditional rendering 👍🏼 - https://lit.dev/docs/templates/conditionals/#conditionals-with-if-statements

Reply:
I think pie-chip has a similar situation in that the slots are in between conditional rendering. So maybe we just need a conditional like `${variant != default ? this.renderIcon() : nothing}`? 😄 

https://github.com/justeattakeaway/pie/blob/main/packages/components/pie-chip/src/index.ts